includes/functions.php

```diff
@@ -103,11 +115,12 @@ function get_site_terms ($blog_id = 0) {
      *
      * @param WP_Term $term
      * @param array $args
+     * @param array $exclude Optional. IDs of categories to exclude. Sites with these categories will be excluded from the list.
      *
      * @return array
      */
-    function get_sites_in_directory_by_term ($term, $args = array()) {
-        $args = wp_parse_args($args, array(
+    function get_sites_in_directory_by_term ($term, $args = array(), $exclude = array() ) {
```

Comment:
Again, same issue as before; you're rewriting the implementation but should be wrapping the core function's interface. :)